workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -202,6 +205,18 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
     coroutineContext.cancel(cause)
   }
 
+  // Call this after we have been passed any workflow instance, in [render] or [snapshot]. It may
+  // have changed and we should check to see if we need to update our cached instances.
+  private fun updateCachedWorkflowInstance(
+    workflow: StatefulWorkflow<PropsT, StateT, OutputT, RenderingT>
+  ) {
+    if (workflow !== cachedWorkflowInstance) {
+      // instance has changed.
+      interceptedWorkflowInstance = interceptor.intercept(workflow, this)
+      cachedWorkflowInstance = workflow
```

Comment:
Why do you update `cachedWorkflowInstance` first in `init`, but second here? Is there some case where `intercept` needs the cached workflow to not be updated yet?

Reply:
Done.